compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1791,12 +1791,44 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
       val sym = dd.symbol
       val isAbstract = isAbstractMethod(dd)
 
+      def isIgnorableDefaultParam: Boolean = {
```

Comment:
I'm really bothered by the duplication with the previous commit here: I realize that a re-structuring almost certainly will mean duplicating computation, but unless there is strong performance evidence, I believe the overall structure should look something like:

```
def isIgnorableDefaultParam = isJSDefaultParam && ! {
  /* exception because of backwards compatibility for ... */
}
```

This would:
- Deduplicate the "real" logic (how we'd want it to be)
- Make the special case explicit.

Personally, I'd even consider simply *always* calculating the target method for the default first (in this and the previous commit). While that might be computationally more expensive, it is much easier to reason about: I don't think anybody thinks in terms of the default accessors, but rather in terms of the methods that have the defaults.

Reply:
I have reduced the duplication somewhat. However, further reducing it seems difficult, because the two logics are actually different, notably for non-native JS classes. I have tried elaborating on those differences in comments, to make it clearer.

Reply:
It's much clearer now. TY